src/components/JourneyComparison/JourneyComparison.tsx

```diff
@@ -33,26 +39,27 @@ const JourneyComparison = (props: ComparisonProps) => {
 
     const baselineTotalDuration = getTotalJourneyDuration(baseline);
     const enhancedTotalDuration = getTotalJourneyDuration(enhanced);
-    const enhancedTotalFraction = 1 - enhancedTotalDuration / baselineTotalDuration;
+    const enhancedTotalFraction =
+        baselineTotalDuration > 0 ? 1 - enhancedTotalDuration / baselineTotalDuration : 0;
     const showDelayRow =
-        enhanced.amenities.includes("electricTrains") &&
-        !baseline.amenities.includes("electricTrains");
-    const showWaitRow = !baseline.reverse;
-    const amenitiesDiff = enhanced.amenities.filter((a) => !baseline.amenities.includes(a));
+        enhanced?.amenities?.includes("electricTrains") &&
+        !baseline?.amenities?.includes("electricTrains");
+    const showWaitRow = !baseline?.reverse;
+    const amenitiesDiff = enhanced?.amenities?.filter((a) => !baseline?.amenities?.includes(a));
 
     return (
         <div className={styles.journeyComparison}>
             <ComparisonRow
                 baseline={
                     <div className="column-header">
                         <div className="header-blip baseline" />
-                        {baseline.scenario.name}
+                        {baseline?.scenario?.name || "Today's commuter rail"}
```

Comment:
This is another place where returning fallback `JourneyInfo` from the server makes our lives simpler, since we can now continue to access `baseline.scenario.name`.